core/client/src/main/java/alluxio/client/file/UnknownLengthFileInStream.java

```diff
@@ -122,6 +124,8 @@ protected BlockInStream createUnderStoreBlockInStream(long blockStart, long leng
     return new UnderStoreBlockInStream(blockStart, Constants.UNKNOWN_SIZE, length, path);
   }
 
+  // TODO(peis): I don't see why this is necessary. Ask Gene to confirm.
```

Comment:
If  the block was fully read, the block should be fully cached or null in the current implementation. I think i can remove all the logic related to mCurrentCacheBytesWritten. Let me know if you see any problem